homeassistant/components/nest/sensor_sdm.py

```diff
@@ -68,6 +72,15 @@ async def async_added_to_hass(self) -> None:
             self._device.add_update_listener(self.async_write_ha_state)
         )
 
+    @property
+    def available(self) -> bool:
+        """Return entity availability."""
+        if ConnectivityTrait.NAME in self._device.traits:
+            trait: ConnectivityTrait = self._device.traits[ConnectivityTrait.NAME]
+            if trait.status == CONNECTIVITY_TRAIT_OFFLINE:
+                return False
+        return super().available
```

Comment:
Move this to a mixin-class `NextAvailability` which you extend `SensorBase` and `ThermostatEntity` with.

Reply:
Created new module availability_mixin.py and class AvailabilityMixin, put logic in here and added to SensorBase and ThermostatEntity

Reply:
I had a challenge in the mixin making all the linter happy about accessing the _device property from the final class.  I did get something to finally make everyone happy.  If there's a better solve, let me know.

Reply:
One suggestion: There's also a `DeviceInfo` class which is basically a small wrapper for interpreting the nest response APIs. While its generic and used for multiple device types, it seems a little more straight forward to add the shared logic there than accessing attributes of an object this is mixed in with.

Reply:
> One suggestion: There's also a `DeviceInfo` class which is basically a small wrapper for interpreting the nest response APIs. While its generic and used for multiple device types, it seems a little more straight forward to add the shared logic there than accessing attributes of an object this is mixed in with.

I like it

I'll add it in there and then implement "def available" in those two classes returning self._device.available